packages/compose/src/hooks/use-instance-id/index.js

```diff
@@ -23,9 +23,9 @@ function createId( object ) {
 /**
  * Provides a unique instance ID.
  *
- * @param {Object} object Object reference to create an id for.
- * @param {string} prefix Prefix for the unique id.
- * @param {string} preferredId Default ID to use.
+ * @param {any} object Object reference to create an id for.
```

Comment:
The type should be `object` instead of `any` (or `unknown`). The `K` type in `WeakMap<K, V>` is supposed to extend `object`. That prevents usages with primitive types that can't be `WeakMap` keys.
```js
useInstanceId( 'hello' );
```
will throw an exception at runtime (invalid value used as weak map key). And if the param's type is `object`, it will be a type error, too.